src/Mesh.cpp

```diff
@@ -213,60 +232,78 @@ Vector3D Mesh::getFormattedReactionRates(std::vector<std::vector<double>> widths
   double offset_y = (root_universe->getMinY() + root_universe->getMaxY()) / 2;
   double offset_z = (root_universe->getMinZ() + root_universe->getMaxZ()) / 2;  
   
+  /* The Lattice defined by user for reaction rates output, it is likely to be 
+     smaller than the whole geometry*/
+  Lattice output_lattice;
   
-  Lattice real_lattice;
-  real_lattice.setNumX(widths_offsets[0].size());
-  real_lattice.setNumY(widths_offsets[1].size());
-  real_lattice.setNumZ(widths_offsets[2].size());
-  real_lattice.setWidths(widths_offsets[0], widths_offsets[1], widths_offsets[2]);
+  output_lattice.setNumX(widths_offsets[0].size());
+  output_lattice.setNumY(widths_offsets[1].size());
+  output_lattice.setNumZ(widths_offsets[2].size());
+  output_lattice.setWidths(widths_offsets[0], widths_offsets[1], 
+                           widths_offsets[2]);
+  
+  /* If no offset coordinates is provided, use the geometry center */
   if(widths_offsets.size() == 3) 
-    real_lattice.setOffset(offset_x, offset_y, offset_z);
+    output_lattice.setOffset(offset_x, offset_y, offset_z);
   else
-    real_lattice.setOffset(widths_offsets[3][0], widths_offsets[3][1], widths_offsets[3][2]);
-  real_lattice.computeSizes();
+    output_lattice.setOffset(widths_offsets[3][0], widths_offsets[3][1], 
+                             widths_offsets[3][2]);
+  output_lattice.computeSizes();
   
+  /* The whole geometry Lattice based on user defined one. To make use of 
+     getFormattedReactionRates function because it's defined on whole geometry*/
```

Comment:
based on the user-defined one.
This new lattice allows to make use of ..., as this routine is defined on the whole geometry

Reply:
done